tokio-postgres/src/replication_client.rs

```diff
@@ -0,0 +1,934 @@
+//! Streaming replication support.
```

Comment:
Rather than having an entirely separate client for this, I would rather just have a `Client::copy_both` method that returns a type implementing `Sink` and `Stream` (like a combination of `CopyInSink` and `CopyOutStream`, and have these more complex interfaces be developed externally initially.

Reply:
The initial motivation for the separate client was to prevent certain methods from being available to the user when in that mode and visa-versa. (I'm speaking on behalf of @jeff-davis here so I hope I'm representing it correctly).

I can certainly make this be a `Client::copy_both` on the client and rely on documentation for the users to know that if they create a client in replication mode and attempt to do normal queries they will get runtime errors.

Reply:
fixed